src/testapp/main.cpp

```diff
@@ -217,68 +206,70 @@ void debug_cli_loop()
                 }
 
                 g_testApp.m_rendererSetup = it->second.m_setup(g_testApp);
-                start_magnum_async();
+                start_magnum_async(argc, argv);
             }
         }
-        else if (command == "help")
-        {
-            debug_print_help();
-        }
-        else if (command == "reopen")
-        {
-            if (magnumOpen)
+        else // Otherwise check all other commands. 
+        { 
+            if (command == "help") 
             {
-                std::cout << "Application is already open\n";
+                print_help();
             }
-            else if ( g_testApp.m_rendererSetup == nullptr )
+            else if (command == "reopen") 
             {
-                std::cout << "No existing scene loaded\n";
+                if (magnumOpen)
+                {
+                    std::cout << "Application is already open" << std::endl;
+                }
+                else if (g_testApp.m_rendererSetup == nullptr)
+                {
+                    std::cout << "No existing scene loaded" << std::endl;
+                }
+                else
+                {
+                    start_magnum_async(argc, argv);
+                }
+                break;
             }
-            else
+            else if (command == "list_pkg") 
             {
-                start_magnum_async();
+                print_resources();
             }
+            else if (command == "exit") 
+            {
+                if (magnumOpen)
+                {
+                    OSP_DECLARE_GET_DATA_IDS(g_testApp.m_renderer.m_sessions[1], TESTAPP_DATA_MAGNUM); // declares idActiveApp
+                    osp::top_get<MagnumApplication>(g_testApp.m_topData, idActiveApp).exit();
 
-        }
-        else if (command == "list_pkg")
-        {
-            debug_print_resources();
-        }
-        else if (command == "exit")
-        {
-            if (magnumOpen)
+                    break;
+                }
+            }
+            else 
             {
-                // Request exit if application exists
-                OSP_DECLARE_GET_DATA_IDS(g_testApp.m_renderer.m_sessions[1], TESTAPP_DATA_MAGNUM); // declares idActiveApp
-                osp::top_get<MagnumApplication>(g_testApp.m_topData, idActiveApp).exit();
+                std::cout << "That command doesn't do anything ._." << std::endl;
             }
-
-            break;
-        }
-        else
-        {
-            std::cout << "that doesn't do anything ._.\n";
         }
     }
 
     g_testApp.clear_resource_owners();
 }
 
-void start_magnum_async()
+void start_magnum_async(int argc, char** argv)
 {
     if (g_magnumThread.joinable())
     {
         g_magnumThread.join();
     }
-    std::thread t([] {
+    std::thread t([&] {
```

Comment:
I agree it makes much more sense to copy them, but for some reason the constructor of "MagnumApplication::Arguments" needs, and stores argc as a reference, it dosent copy in the value, that code is within magnum so solving that issue is a bit of an undertaking.

Reply:
huh, I see. They can still be stored by value as part of the lambda with `[argc, argv]`. This would stay valid for the entire lifetime of this thread (until the window is closed). It's just odd that this would break if argc or argv didn't have a global lifetime.

less important information:
There's no mechanism to know if a variable global or not, or anything about its lifetime (rust can do this with the `'static` lifetime specifier, but this is not rust of course).

Reply:
On MSVC I cannot successfully compile while using `[argc, argv]`, because the lambda makes argc into a const int, which you cant make a reference too. I made the solution slightly better by making it so only argc is captured by reference while argv is copied by value.

I am going to make a pull request in the magnum repo that makes the arguments class/struct take in by value instead of by reference (Because its really weird that it needs a reference), but in the short term I believe that the current solution should work, but if you would like, I can revert it back to the previous solution where both argc and argv are stored as globals.

Reply:
Forgot that lambda captures are one of the few things in C++ that are const by default. The lambda can be non-const by adding the `mutable` keyword:  `[argc, argv] mutable { ... }`

Reply:
I did not know about that, I just implemented it and all seems to be working well